### DISCUSSION TITLE
Troubleshooting - Adding ssh-key to the ssh-agent

### DISCUSSION THREAD

**Discussion by inswo** [community/community#39586] [New to GitHub] [answered]
Hi everyone,

I'm a newbie in coding and I might need some help.

I'm working from a Macbook Pro - Mac OS Monterey 12.6.1 and I even though I followed each step from this article (https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent), when I type "$ touch ~/.ssh/config" on my terminal to create the file, my terminal return "No such file or directory".

Thanks a lot !

**Comment by DSDmark** [comment, marked as answer]
Hi @inswo,

I think you are trying to get your **ssh** key, right.

Try this these steps:-
![2022-11-20_23-31](https://user-images.githubusercontent.com/85744816/202910930-a602bfaa-129f-411b-a2ab-8d684a26a8a6.png)

**Reply by inswo** [reply]
Hi @DSDmark ,
Thanks for your answer !
I generated a SSH key. However, I can't add the ssh key to the ssh-agent.

**Reply by DSDmark** [reply]
@inswo,
That command not working 
```bash
eval "$(ssh-agent -s)"
 ```
If that case. Can you add **screenshot** or **error** which you are facing

**Reply by inswo** [reply]
<img width="555" alt="Screenshot 2022-11-20 at 16 46 17" src="https://user-images.githubusercontent.com/97383958/202911651-793d828f-c634-4cf2-b0ba-10f03ca4aaea.png">

**Reply by inswo** [reply]
@DSDmark , here's the scrrenshot, thanks !

**Reply by DSDmark** [reply]
@inswo,
ok, I think when you generated **ssh** key you give wrong path.
Try this:- ```find -name ".ssh"```

**Reply by DSDmark** [reply]
it's show the path of your directory where that file is

**Reply by inswo** [reply]
@DSDmark 
Insoo@MacBook-Pro-de-Insoo ~ % find -name ".ssh"
find: illegal option -- n
usage: find [-H | -L | -P] [-EXdsx] [-f path] path ... [expression]
       find [-H | -L | -P] [-EXdsx] -f path [path ...] [expression]

**Reply by DSDmark** [reply]
@inswo,
I don't know about **mac** which terminal environment it's using. But, I guess you could try this. 
> ```find . -name '.ssh'``` or that ```find / -name '.ssh'```

**Reply by inswo** [reply]
@DSDmark , 
I dont know if it can helps but when I type 'ls' command, I can see the key file.
The command `find . -name '.ssh'` keeps returning 'OPeration not permitted'

**Reply by DSDmark** [reply]
@inswo,
```ls``` Or did you mean ```ls -la```.
Can you share screenshot?

**Reply by inswo** [reply]
<img width="467" alt="Screenshot 2022-11-20 at 17 31 12" src="https://user-images.githubusercontent.com/97383958/202913842-a66f9b2d-ab56-4160-a146-f1909333a6f6.png">
@DSDmark

**Reply by DSDmark** [reply]
I'm not sure. If it's your **ssh** key. Then, did you try this:-```ssh-add ./git-key```

**Reply by inswo** [reply]
It looks like it worked, what should I do next ?

**Reply by DSDmark** [reply]
@inswo,
I guess, it's done,
here [>>>](https://youtu.be/dQw4w9WgXcQ)

**Reply by inswo** [reply]
@DSDmark ,
When I type ssh-add --apple-use-keychain ~/.ssh/id_ed25519 it shows "No such file or directory"

**Reply by DSDmark** [reply]
@inswo,
It's not done yet?
I guess because you incidentally change your file name. Not sure about that.

**Reply by DSDmark** [reply]
I never generate key in mac, so maybe I don't know

**Reply by inswo** [reply]
I might be wrong but I believe the file name and the id is different ?

**Reply by DSDmark** [reply]
maybe

**Reply by inswo** [reply]
Thanks a lot !

**Comment by ahmdsalhh** [comment]
ahmdsalhhzamalfatqy@gmail.com
